NYPLAudiobookToolkit/Network/OpenAccessDownloadTask.swift

```diff
@@ -120,12 +132,13 @@ final class OpenAccessDownloadTask: DownloadTask {
     /// to the url of the actual asset to download.
     private func downloadAssetForRBDigital(toLocalDirectory localURL: URL) {
 
-        let task = URLSession.shared.dataTask(with: url) { (data, response, error) in
+        let task = URLSession.shared.dataTask(with: url) { [weak self] (data, response, error) in
 
             guard let data = data,
                 let response = response,
                 (error == nil) else {
                 ATLog(.error, "Network request failed for RBDigital partial file. Error: \(error!.localizedDescription)")
+                self?.removeDownloadTimer()
```

Comment:
I don't know if we should, but should we remove the timer in all cases? Basically I'm worried if there are cases where the timer is left going forever. Ideally we want to see a clear correlation between timer start and stop.

Reply:
I am thinking of creating two functions, `notifyDelegateDownloadTaskFailed(error:)` and `notifyDelegateDownloadTaskReadyForPlayback()`. Within these two functions, we call the delegate methods of download fails and completes and remove the timer. We should end up reaching these two functions in all the cases, except when the download task time out and we manually cancel the download task as well as the timer.

Reply:
sounds good -- but if the task times out, isn't that a fail too?

Reply:
Right, but the time out is handled by the timer within `AudiobookNetworkService` in our case, and we then call `downloadTask.cancel()` from there and that's where we remove the timer.